tower/src/filter/mod.rs

```diff
@@ -72,6 +72,14 @@ impl<T, U> Filter<T, U> {
     pub fn layer(predicate: U) -> FilterLayer<U> {
         FilterLayer::new(predicate)
     }
+
+    /// Check a `Request` value against this filter's predicate.
+    pub fn check<Request>(&mut self, request: Request) -> Result<U::Request, BoxError>
```

Comment:
only nit I have is we have a lot of `Request` in here, maybe change the generic to be `R`?